source/bloodflowmodel/pressure_flow_solver.py

```diff
@@ -49,8 +50,101 @@ def _solve_flow(self, flownetwork):
+
+        # if we are in the iterative case of Rasmussen
+        if self._PARAMETERS["iterative_routine"] == 4:
+            # at the first iteration is necessary to set the flow_threshold criteria
+            if flownetwork.iteration == 1:
+                flownetwork.rasmussen_flow_threshold = 1 * 10 ** (int("{:e}".format(flownetwork.flow_convergence_criteria).split('e')[1]) - 8)
+            flownetwork.flow_convergence_criteria = max(abs(abs(flow_rate) - abs(flownetwork.flow_rate)))
+
+        flownetwork.flow_rate = flow_rate
+
+
+def flow_berg(flownetwork, flow_rate):
+    # ||Q^n - Q^n-1||
+    flow_norm = np.linalg.norm(abs(abs(flow_rate) - abs(flownetwork.flow_rate)))
+    # Q_i^n implemented as the sum pressure inlet
+    # inflow: are all the inflow values sum
+    return flow_norm / flownetwork.inflow
+
+
+def pressure_berg(flownetwork, pressure):
+    # ||P^n - P^n-1||
+    pressure_norm = np.linalg.norm(abs(pressure - flownetwork.pressure))
+    # P_i^n implemented as the average pressure inlet
+    # inflow_pressure_node: are all the inflow node
+    average_inlet_pressure = np.average(flownetwork.pressure[flownetwork.inflow_pressure_node])
+    flownetwork.average_inlet_pressure.append(average_inlet_pressure)
+
+    return pressure_norm / average_inlet_pressure
+
+
+def _berg_assistance(flownetwork):
+    from source.bloodflowmodel.flow_balance import dict_for_families_total
+    inflow_pressure_node, inflow = [], 0
+    # find the parent and daughters of every node
+    families_dict = dict_for_families_total(flownetwork)
+    for key, values in families_dict.items():
+        if key in flownetwork.boundary_vs:
+            if values['dgs']:
+                # for the flow is necessary to do a bit more preprocessing
+                # if par = [ ] it means that is a boundary node and the flow is exact the one needed
+                if values['par'] == []:
+                    inflow += sum(abs(flownetwork.flow_rate[key]) for key in values['dgs'])
+                    inflow_pressure_node.append(key)
+
+                else:
+                    # if the par vessel exists, it is possible that our inflow is smaller
+                    # that the one this goes to the daughters one
+                    flow_par = sum(abs(flownetwork.flow_rate[key]) for key in values['par'])
+                    flow_dgs = sum(abs(flownetwork.flow_rate[key]) for key in values['dgs'])
+                    if flow_dgs > flow_par:
+                        inflow += abs(flow_dgs - flow_par)
+                        inflow_pressure_node.append(key)
+    flownetwork.inflow_pressure_node, flownetwork.inflow = inflow_pressure_node, inflow
+
+
+def _update_low_flow(flownetwork, flow_rate):
+    # Update flow rate based on the zero-flow threshold
+    flow_rate = np.where(np.abs(flow_rate) < flownetwork.zeroFlowThreshold, 0, flow_rate)
+    return flow_rate
+
+
+def set_low_flow_threshold(flownetwork, local_balance):
```

Comment:
Can we discuss this function again in person? 
Thanks!
Note for me: This function is also used in the flow_balance.py